internal/backends/nodejs/nodejs.go

```diff
@@ -414,13 +414,28 @@ var BunBackend = api.LanguageBackend{
 	},
 	ListSpecfile: nodejsListSpecfile,
 	ListLockfile: func() map[api.PkgName]api.PkgVersion {
-		// when bun executes the lockfile it prints out a Yarn specfile
-		yarnSpec, err := exec.Command("bun", "bun.lockb").Output()
+		hashString, err := exec.Command("bun", "pm", "hash-string").Output()
 		if err != nil {
-			util.Die("bun.lockb: %s", err)
+			util.Die("bun pm hash-string: %s", err)
 		}
 
-		return parseYarnSpec(yarnSpec)
+		// name could be:
+		// - @scope/name
+		// - scope/name
+		// - name
+		// version could be:
+		// - 1.2.3
+		// - 1.2.3-beta.1
+		// - 1.2.3-beta.1-build.1
+		r := regexp.MustCompile(`(?m)^(@?[^@ \n]+)@([0-9]+\.[0-9]+\.[0-9]+(-.+)?)$`)
```

Comment:
When the version is a git dependency, github dependency, an aliased npm package, a tarball url, a workspace, or a symlink, the format in the version specifier changes. Instead, I suggest matching everything after the last`"@"` up until the newline character

For example:

<img width="494" alt="image" src="https://user-images.githubusercontent.com/709451/230685443-e49a6c5c-c102-40b5-8749-1bc16b0f171d.png">

Reply:
ah, good catch - fixed in latest commit